src/main/java/com/google/opensesame/projects/ProjectTag.java

```diff
@@ -0,0 +1,43 @@
+package com.google.opensesame.projects;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.kohsuke.github.GHRepository;
+
+public class ProjectTag {
+  /**
+   * Creates a list of ProjectTags from a GitHub repository's topics and primary language.
+   *
+   * @param repository The GitHub repository.
+   * @return Returns the list of ProjectTags.
+   * @throws IOException
+   */
+  public static List<ProjectTag> fromRepository(GHRepository repository) throws IOException {
+    List<String> topics = repository.listTopics();
+    String primaryLanguage = repository.getLanguage();
+
+    ArrayList<ProjectTag> projectTags = new ArrayList<ProjectTag>(topics.size() + 1);
+    projectTags.add(new ProjectTag(primaryLanguage));
+    for (String topic : topics) {
+      projectTags.add(new ProjectTag(topic));
+    }
+
+    return projectTags;
+  }
+
+  /**
+   * Creates a new ProjectTag.
+   *
+   * @param tagText
+   */
+  public ProjectTag(String tagText) {
+    this.tagText = tagText;
+  }
+
+  private final String tagText;
+
+  public String getTagText() {
+    return tagText;
+  }
+}
```

Comment:
Let's have a video call to talk about this file because if we can significantly simplify or even delete this logic it could make everyone's lives easier. However, there might be something I'm missing.

Reply:
I made the unfortunate decision of removing a comment at the top of this class very recently that gave a bit more context as to the future plans haha. I intend to add a String type to each tag in the near future, so that languages and topics can be easily distinguished on the frontend via color-coding.

At that point it will still only be a wrapper class of two strings, so I wonder if you feel that the class should be removed?

Reply:
I would like to know what came of this conversation. I think it is worth it to have a tag type for color coordination reasons. Maybe make a simple class of type ```Tag``` that's a wrapper for a ```String```.

Reply:
@Obinnabii Let's talk during sync! We came to a good conclusion with this.